File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.java
class Solution {
    int res;
    String[] arr = new String[]{"aya", "ye", "woo", "ma"};
    public void func(String target, String str, int lev, String lastAdd) {
        if(lev >= 10) return;
        if(str.equals(target)) {
            res++;
            return;
        }
        else {
            for(int i = 0; i < arr.length; i++) {
                if(lastAdd.equals(arr[i])) continue;
                func(target, str + arr[i], lev + 1, arr[i]);
            }
        }
    }
    
    public int solution(String[] babbling) {
        for(String str : babbling) {
            func(str, "", 0, "");
        }
        return res;
    }
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    answer = 0
    words = ["aya", "ye", "woo", "ma"]
    banned_words = ["ayaaya", "yeye", "woowoo", "mama"]
    for actual_word in babbling:
        for banned_word in banned_words:
            actual_word = actual_word.replace(banned_word, "x")
        for word in words: 
            actual_word = actual_word.replace(word, " ")
        
        if len(actual_word.strip()) == 0:
            answer = answer + 1
        
    return answer